### ISSUE TITLE
`s3-controller` fails to sync buckets created in regions which do not support Transfer Acceleration

### ISSUE THREAD

**Issue by Ross-Duckworth-Pharos** [aws-controllers-k8s/community#1206]
**Describe the bug**
If you create an s3 bucket resource in a region such as `af-south-1`, then the `s3-controller` will fail to sync its specification because it is unconditionally calling the S3 API `GetBucketAccelerateConfiguration` which is not allowed for certain regions, as documented here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/transfer-acceleration.html. This aborts the rest of the specification syncing.

Issue lies with this code: https://github.com/aws-controllers-k8s/s3-controller/blob/2153e6d2cf34cd9d68cf0f27dbc3c3610644a7e3/pkg/resource/bucket/hook.go#L297
Suggest it should trap the `MethodNotAllowed` error and continue.

**Steps to reproduce**
1. Apply the following `bucket` yaml via `kubectl`:
```
apiVersion: s3.services.k8s.aws/v1alpha1
kind: Bucket
metadata:
  name: test-bucket
  annotations:
    services.k8s.aws/region: af-south-1
spec:
  name: test-bucket
```

2. Inspect the `bucket` resource events in Kubernetes, and you will see an error like the following:
_MethodNotAllowed: The specified method is not allowed against this resource._

3. You can also reproduce via the AWS CLI:
_aws s3api get-bucket-accelerate-configuration --bucket test-bucket --region af-south-1_

**Expected outcome**
That the created `bucket` resource in Kubernetes is successfully synced with the S3 bucket.

**Environment**

* Kubernetes version
1.21.5
* Using EKS (yes/no), if so version? 
Yes, `eks.4`.
* AWS service targeted (S3, RDS, etc.)
S3

**Comment by RedbackThomson** [comment]
I searched through as much documentation as I could find for each S3 property, and this seems to be the only one that has different requirements for opt-in regions. Please lmk if there are other services that have special needs for af-south-1! This is a big thing I hadn't considered.

**Comment by RedbackThomson** [comment]
Fixed with https://github.com/aws-controllers-k8s/s3-controller/pull/72

Unfortunately this fix didn't make it into `v0.0.16` of the s3-controller. Let me know if you urgently need this and I can cut a `v0.0.17` for it.
